### DISCUSSION TITLE
Querying large number of rows in C++.

### DISCUSSION THREAD

**Discussion by peter-facko** [questdb/questdb#4970] [Ideas]
Hello,

I am using C++ in my project. I need to process hundreds of millions of rows from a single query.

I read the documentation about cursors but it seems that `libpqxx` implements cursors using `DECLARE` which is not supported.

The only other solution I came up with is to use `LIMIT` with lower and upper bounds. Is there any better way I am missing? It seems weird that C++ is the language with which I cannot natively process large amounts of data from QuestDB.

**Comment by peter-facko** [comment]
`COPY query TO STDOUT` seems like an interesting feature of PostgreSQL to consider.

**Comment by ideoma** [comment]
You can now use Qwp protocol to read the data. It's much more effective than any other export/query protocol 
https://questdb.com/docs/connect/clients/c-and-cpp/
